src/app/pages/office/cat-b/office.cat-b.page.ts

```diff
@@ -0,0 +1,36 @@
+import { Component, OnInit } from '@angular/core';
+import { NavController } from '@ionic/angular';
+import { TestCategory } from '@dvsa/mes-test-schema/category-definitions/common/test-category';
+import { Router } from '@angular/router';
+import { RouteByCategoryProvider } from '@providers/route-by-category/route-by-category';
+import { JOURNAL_PAGE } from '../../page-names.constants';
+
+@Component({
+  selector: 'app-office.cat-b',
+  templateUrl: './office.cat-b.page.html',
+  styleUrls: ['./office.cat-b.page.scss'],
+})
+export class OfficeCatBPage implements OnInit {
+
+  constructor(
+    private navController: NavController,
+    public routeByCat: RouteByCategoryProvider,
+    private router: Router,
+  ) { }
+
+  ngOnInit() {
+  }
+
+  navigateBack(): void {
+    this.navController.back();
+  }
+
+  async navigateForward(): Promise<void> {
+    await this.routeByCat.navigateToPage('REKEY_REASON_PAGE', TestCategory.B);
+  }
+
+  async navigateJournal(): Promise<void> {
+    await this.router.navigate([JOURNAL_PAGE]);
```

Comment:
are we able to do remove the use of router here and change to this?

 await this.routeByCat.navigateToPage(JOURNAL_PAGE);

Reply:
no - navigate to page doesnt effect the routes already specified in app routing, only adds to them dynamically. so dashboard, login, journal and test centre journal still need to use router.navigate directly.